### PR TITLE
[8_1] add icon for mogan under windows

### PR DESCRIPTION
**Pull request by jingkaimori**
rc was introduced from separate target, and it will be enabled only under windows environment.

### CODE REVIEW

**Review by Yufeng-shen — APPROVED** [review]
looks good to me

File: xmake.lua [hunk 1]
```diff
@@ -656,6 +656,20 @@ option("libdl") do
...
+    target("windows_icon") do
+        set_version(XMACS_VERSION)
+        set_kind("object")
+        add_configfiles("packages/windows/resource.rc.in", {
+            filename = "resource.rc"
+        })
+        add_configfiles("packages/windows/TeXmacs.ico", {
+            onlycopy = true
+        })
+        add_files("$(buildir)/resource.rc")
```
Comment (darcy-shen) [inline xmake.lua:669]:
```
warning: .\xmake.lua:669: cannot match add_files("build/resource.rc") in target(windows_icon)
```

Comment (darcy-shen) [reply xmake.lua:669]: https://github.com/xmake-io/xmake/issues/3229
another problem is: `error: ...mdir\core\sandbox\modules\import\core\base\scheduler.lua:56: cannot get program for mrc`

Comment (darcy-shen) [reply xmake.lua:669]: `rc.exe`是需要安装Windows SDK，这对于开发者来说是不需要的。这块应该放在打包的阶段

**Review by darcy-shen — CHANGES_REQUESTED** [review]
windows_icon should be build when packaging

File: xmake.lua (hunk 1 above)
Comment (darcy-shen) [reply xmake.lua:669]: https://github.com/xmake-io/xmake/issues/3942

不需要安装 Windows SDK，只需要把mingw加到path里面就可以了

**Review by darcy-shen — APPROVED** [review]
LGTM now

**Pull request merged**